lasagne/layers/conv.py

```diff
@@ -794,8 +793,9 @@ def convolve(self, input, **kwargs):
             subsample=self.stride, border_mode=border_mode,
             filter_flip=not self.flip_filters)
         output_size = self.output_shape[2:]
-        if any(s is None for s in output_size):
-            output_size = self.get_output_shape_for(input.shape, True)[2:]
+        if (not isinstance(self.output_size, T.TensorVariable) and
+                any(s is None for s in output_size)):
+            output_size = self.get_output_shape_for(input.shape)[2:]
```

Comment:
> I modified your suggestion because when self.output_size is None the any(s is None for s in output_size) would fail

No, why? There's `output_size = self.output_shape[2:]` before, not `output_size = self.output_size`. You can use my code as-is.